control-plane/connect-inject/peering_acceptor_controller.go

```diff
@@ -52,16 +54,32 @@ func (r *PeeringAcceptorController) Reconcile(ctx context.Context, req ctrl.Requ
 	// If the PeeringAcceptor resource has been deleted (and we get an IsNotFound
 	// error), we need to delete it in Consul.
 	if k8serrors.IsNotFound(err) {
-		r.Log.Info("PeeringAcceptor was deleted, deleting from Consul", "name", req.Name, "ns", req.Namespace)
-		if err := r.deletePeering(ctx, req.Name); err != nil {
-			return ctrl.Result{}, err
-		}
+		r.Log.Info("PeeringAcceptor resource not found. Ignoring.", "name", req.Name, "ns", req.Namespace)
 		return ctrl.Result{}, nil
 	} else if err != nil {
 		r.Log.Error(err, "failed to get PeeringAcceptor", "name", req.Name, "ns", req.Namespace)
 		return ctrl.Result{}, err
 	}
 
+	if acceptor.GetDeletionTimestamp().IsZero() {
```

Comment:
This will add the finalizer to any resource that doesn't already have it right? Not just recently deleted resources? Could we add a comment for what this case is describing?

Reply:
Let me know if the comment makes sense!

Reply:
Yep its great, tysm!!